config/config_helpers.go

```diff
@@ -301,6 +307,108 @@ func getAWSAccountID(include *IncludeConfig, terragruntOptions *options.Terragru
 	return *identity.Account, nil
 }
 
+// Return the output from the state of another module, managed by terragrunt. This function will parse the provided
+// terragrunt config and extract the desired output from the remote state. Note that this will error if the targetted
+// module hasn't been applied yet. If you omit the output name, this will return a map of all the outputs from the
+// module.
+func getTerragruntOutput(params []string, include *IncludeConfig, terragruntOptions *options.TerragruntOptions) (*cty.Value, error) {
+	// args check: make sure the expected args are passed
+	numParams := len(params)
+	if numParams == 0 {
+		return nil, errors.WithStackTrace(EmptyStringNotAllowed("the target terragrunt config parameter"))
+	}
+	if numParams > 2 {
+		return nil, errors.WithStackTrace(WrongNumberOfParams{Func: "get_output", Expected: "1 or 2", Actual: numParams})
+	}
+
+	// target config check: make sure the target config exists
+	cwd := filepath.Dir(terragruntOptions.TerragruntConfigPath)
+	targetConfig := params[0]
+	if !filepath.IsAbs(targetConfig) {
+		targetConfig = util.JoinPath(cwd, targetConfig)
+	}
+	if util.IsDir(targetConfig) {
+		targetConfig = util.JoinPath(targetConfig, DefaultTerragruntConfigPath)
+	}
+	if !util.FileExists(targetConfig) {
+		return nil, errors.WithStackTrace(TerragruntOutputConfigNotFound{Path: targetConfig})
+	}
+
+	jsonBytes, err := runTerragruntOutputJson(terragruntOptions, targetConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	outputMap, err := terraformOutputJsonToCtyValueMap(targetConfig, jsonBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	if numParams == 2 {
+		// When indexing a single output, we can return the value directly after parsing the json.
+		tmp := outputMap[params[1]]
```

Comment:
Will this `panic` if `outputMap` is missing the key? Should the two-return value version of map lookup be used instead?

Reply:
This is no longer relevant as the indexing for specific outputs is now done in HCL2, not golang.